workflow/executor/resource.go

```diff
@@ -20,17 +20,51 @@ import (
 )
 
 // ExecResource will run kubectl action against a manifest
-func (we *WorkflowExecutor) ExecResource(action string, manifestPath string) (string, string, error) {
-	isDelete := action == "delete"
+func (we *WorkflowExecutor) ExecResource(action string, manifestPath string, flags []string) (string, string, error) {
+	args, err := we.getKubectlArguments(action, manifestPath, flags)
+	if err != nil {
+		return "", "", err
+	}
+
+	cmd := exec.Command("kubectl", args...)
+	log.Info(strings.Join(cmd.Args, " "))
+
+	out, err := cmd.Output()
+	if err != nil {
+		exErr := err.(*exec.ExitError)
+		errMsg := strings.TrimSpace(string(exErr.Stderr))
+		return "", "", errors.New(errors.CodeBadRequest, errMsg)
+	}
+	if action == "delete" {
+		return "", "", nil
+	}
+	obj := unstructured.Unstructured{}
+	err = json.Unmarshal(out, &obj)
+	if err != nil {
+		return "", "", err
+	}
+	resourceName := fmt.Sprintf("%s.%s/%s", obj.GroupVersionKind().Kind, obj.GroupVersionKind().Group, obj.GetName())
+	log.Infof("%s/%s", obj.GetNamespace(), resourceName)
+	return obj.GetNamespace(), resourceName, nil
+}
+
+func (we *WorkflowExecutor) getKubectlArguments(action string, manifestPath string, flags []string) ([]string, error) {
 	args := []string{
 		action,
 	}
 	output := "json"
+
+	isDelete := action == "delete"
 	if isDelete {
```

Comment:
Looks like we don't need `isDelete`
```suggestion
	if action == "delete" {
```